api/controllers/user.js

```diff
@@ -119,50 +117,47 @@ router.get('/get/:userId', function edituserById (req, res, next) {
  *
  */
 
-router.post('/update', function updateUser (req, res, next) {
+router.post("/update", function updateUser(req, res, next) {
   console.log(req.body.user);
   if (req.body.secretToken != process.env.SECRET_TOKEN) {
-    res.status(401)
-      .json({
-        status: 'unauthorized',
-        message: "Call didn't pass in right secret token"
-      });
+    res.status(401).json({
+      status: "unauthorized",
+      message: "Call didn't pass in right secret token"
+    });
   } else {
     // See if the user exists.
-    var userId = req.body.user.user_id
-    var name = req.body.user.name
-    db.one('select * from users where id = $1', userId)
-      .then(function (data) {
+    let userId = req.body.user.user_id;
+    let name = req.body.user.name;
+    db
+      .one("select * from users where id = $1", userId)
+      .then(function(data) {
         // If user exists, do an update
-        db.none('update users set name=$1 where id=$4',
-          [name, userId])
-          .then(function () {
-            res.status(200)
-              .json({
-                status: 'success',
-                message: 'Updated user'
-              });
+        db
+          .none("update users set name=$1 where id=$4", [name, userId])
+          .then(function() {
+            res.status(200).json({
+              status: "success",
+              message: "Updated user"
+            });
           })
-          .catch(function (err) {
+          .catch(function(err) {
             return next(err);
           });
       })
-      .catch(function (err) {
-        db.none('insert into users(name)' +
-            'values(${name})',
-          req.body.user)
-          .then(function () {
-            res.status(200)
-              .json({
-                status: 'success',
-                message: 'Inserted user'
-              });
+      .catch(function(err) { /* this feels like it assumes what the error was */
```

Comment:
This feels like an assumption in the code that I worry could mask other error types.  IIUC, you're assuming that the `catch` happens because there is no result to the `one` call?